show-viewshed-from-point-in-scene/src/main/java/com/esri/arcgismaps/sample/showviewshedfrompointinscene/components/SceneViewModel.kt

```diff
@@ -0,0 +1,137 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.showviewshedfrompointinscene.components
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.AndroidViewModel
+import com.arcgismaps.analysis.LocationViewshed
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ArcGISScene
+import com.arcgismaps.mapping.ArcGISTiledElevationSource
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Surface
+import com.arcgismaps.mapping.layers.ArcGISSceneLayer
+import com.arcgismaps.mapping.view.AnalysisOverlay
+import com.arcgismaps.mapping.view.Camera
+import com.arcgismaps.mapping.view.OrbitLocationCameraController
+import com.esri.arcgismaps.sample.showviewshedfrompointinscene.R
+
+class SceneViewModel(private val application: Application) : AndroidViewModel(application) {
+    // set the SceneView mutable stateflow
+    val sceneViewState = SceneViewState()
+
+    private var viewShed: LocationViewshed
+
+    // initialize location viewshed parameters
+    private val initHeading = 82
+    private val initPitch = 60
+    private val initHorizontalAngle = 75
+    private val initVerticalAngle = 90
+    private val initMinDistance = 0
+    private val initMaxDistance = 1500
+
+    init {
+        // create a surface for elevation data
+        val surface = Surface().apply {
+            elevationSources.add(ArcGISTiledElevationSource(application.getString(R.string.elevation_service)))
+        }
+
+        // create a layer of buildings
+        val buildingsSceneLayer =
+            ArcGISSceneLayer(application.getString(R.string.buildings_layer))
+
+        // create a scene and add imagery basemap, elevation surface, and buildings layer to it
+        val buildingsScene = ArcGISScene(BasemapStyle.ArcGISImagery).apply {
+            baseSurface = surface
+            operationalLayers.add(buildingsSceneLayer)
+        }
+
+        val initLocation = Point(-4.50, 48.4, 1000.0)
+        // create viewshed from the initial location
+        viewShed = LocationViewshed(
+            initLocation,
+            initHeading.toDouble(),
+            initPitch.toDouble(),
+            initHorizontalAngle.toDouble(),
+            initVerticalAngle.toDouble(),
+            initMinDistance.toDouble(),
+            initMaxDistance.toDouble()
```

Comment:
Named arguments would be helpful here:
```suggestion
        viewShed = LocationViewshed(
            location = initLocation,
            heading = initHeading.toDouble(),
            pitch = initPitch.toDouble(),
            horizontalAngle = initHorizontalAngle.toDouble(),
            verticalAngle = initVerticalAngle.toDouble(),
            minDistance = initMinDistance.toDouble(),
            maxDistance = initMaxDistance.toDouble()
```